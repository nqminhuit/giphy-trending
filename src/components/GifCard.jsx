import React, { useContext } from "react";
import { useHistory } from "react-router";
import GifAuthorInfo from "./GifAuthorInfo.jsx";
import GifMeta from "./GifMeta.jsx";

export const GifMetaDataContext = React.createContext(null);

export default function GifCard() {
  const history = useHistory();
  const { imgId, imgTitle, imgSrc, imgOrgriginalSrc } = useContext(GifMetaDataContext);

  return (
    <>
      {imgSrc && (
        <div className="w-95 m-auto font-size-small">
          <div className="d-flex flex-column bg-white shadow gif-card-fixed-height">
            <div className="d-flex flex-grow-1 mx-auto overflow-hidden">
              <img
                className="img-fluid gif-card-max-height p-2 cursor-pointer align-self-center fit-contain hover-scale transition-eio-3"
                src={imgSrc}
                alt={imgTitle}
                onClick={() => history.push(`view/${imgId}`, { gifOriginUrl: imgOrgriginalSrc })}
              />
            </div>
            <GifMeta />
          </div>
          <GifAuthorInfo />
        </div>
      )}
    </>
  );

}
