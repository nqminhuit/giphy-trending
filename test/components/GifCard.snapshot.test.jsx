import React from "react";
import renderer from "react-test-renderer";
import GifCard, { GifMetaDataContext } from "../../src/components/GifCard.jsx";

it("should render <GifCard />", () => {
  const component = renderer.create(
    <GifMetaDataContext.Provider value={{
      imgId: "38e10e68-1749-426e-8988-f1d578ee87cf",
      imgSrc: "https://www.gif.fixed_height.webp",
      imgOrgriginalSrc: "https://www.gif.original/url",
      imgTitle: "title",
      numView: 10_000,
      numComment: 100,
      numLove: 1000,
      authorAvatarUrl: "https://www.author.avatar.url",
      authorProfileUrl: "https://www.author.profile.url",
      authorUsername: "username",
    }}>
      <GifCard />
    </GifMetaDataContext.Provider>
  ).toJSON();
  expect(component).toMatchSnapshot();
});
