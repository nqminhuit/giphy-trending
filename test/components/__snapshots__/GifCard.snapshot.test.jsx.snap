// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <GifCard /> 1`] = `
<div
  className="w-95 m-auto font-size-small"
>
  <div
    className="d-flex flex-column bg-white shadow gif-card-fixed-height"
  >
    <div
      className="d-flex flex-grow-1 mx-auto overflow-hidden"
    >
      <img
        alt="title"
        className="img-fluid gif-card-max-height p-2 cursor-pointer align-self-center fit-contain hover-scale transition-eio-3"
        onClick={[Function]}
        src="https://www.gif.fixed_height.webp"
      />
    </div>
    <div
      className="d-flex justify-content-between px-1 mb-1"
    >
      <span
        className="cursor-pointer text-black-50 text-hover-royalblue"
      >
        <i
          className="bi bi-paperclip"
        />
      </span>
      <div>
        <span
          className="me-3 cursor-pointer text-black-50 text-hover-royalblue"
          onClick={[Function]}
        >
          <i
            className="bi bi-eye-fill"
          >
            10000
          </i>
        </span>
        <span
          className="me-3 cursor-pointer text-black-50 text-hover-royalblue"
        >
          <i
            className="bi bi-chat-quote-fill"
          >
            100
          </i>
        </span>
        <span
          className="pe-1 cursor-pointer text-black-50 text-hover-royalblue"
        >
          <i
            className="bi bi-heart-fill"
          >
            1000
          </i>
        </span>
      </div>
    </div>
  </div>
  <div
    className="mt-2 ms-2"
  >
    <a
      className="text-decoration-none fw-bold cl-royalblue"
      href="https://www.author.profile.url"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="giphy author image"
        className="max-height-1rem rounded-circle me-1"
        src="https://www.author.avatar.url"
      />
      username
    </a>
  </div>
</div>
`;
